app/gui/src/project-view/components/MarkdownEditor/FrontMatter.vue

```diff
@@ -0,0 +1,117 @@
+<script setup lang="ts">
+import type { DocumentationMetadata } from '@/components/DocumentationPanel/metadata'
+import DocsTag from '@/components/MarkdownEditor/DocsTag.vue'
+import { computed, ref, watch } from 'vue'
+
+const { metadata } = defineProps<{
+  metadata: DocumentationMetadata
+}>()
+
+interface Tag {
+  label: string
+  value?: string
+  class?: string
+  style?: { [key: string]: string | number | undefined }
+}
+
+const tags = computed<Tag[]>(() => {
+  return [
+    ...(metadata.group ?
+      [
+        {
+          label: metadata.group.toLowerCase().replace('standard.base.', ''),
+          style: {
+            'background-color': 'var(--enso-docs-group-color, #5f5e5e)',
+            color: '#fff',
+          },
+        },
+      ]
+    : []),
+    ...(metadata.unstable ?
+      [
+        {
+          label: 'unstable',
+          style: {
+            'background-color': '#e85252',
+            color: '#fff',
+          },
+        },
+      ]
+    : []),
+    ...(metadata.advanced ?
+      [
+        {
+          label: 'advanced',
+          style: {
+            'background-color': '#e89d51',
+            color: '#fff',
+          },
+        },
+      ]
+    : []),
+    ...(metadata.deprecated ?
+      [
+        {
+          label: 'deprecated',
+          style: {
+            'background-color': '#e89d51',
+            color: '#fff',
+          },
+        },
+      ]
+    : []),
+    ...(metadata.private ? [{ label: 'private' }] : []),
+    ...(metadata.aliases ?? []).map((value) => ({ label: value })),
+    ...(metadata.macros ?? []).map(({ description }) => ({
+      label: description,
+    })),
+    ...(metadata.added ? [{ label: 'added', value: metadata.added }] : []),
+    ...(metadata.modified ? [{ label: 'modified', value: metadata.modified }] : []),
+    ...(metadata.removed ? [{ label: 'removed', value: metadata.removed }] : []),
+    ...(metadata.upcoming ? [{ label: 'upcoming', value: metadata.upcoming }] : []),
+  ]
+})
+
+// === Show/hide excess tags. ===
+
+const containerRef = ref<HTMLDivElement>()
+const parentOffset = computed(() => containerRef.value?.offsetTop ?? 0)
+
+const hiddenTags = ref(0)
+const tagWasHidden = () => (hiddenTags.value += 1)
+const someTagsAreHidden = computed(() => hiddenTags.value > 0)
+
+const showAllTags = ref(false)
+const showEllipsis = computed(() => someTagsAreHidden.value && !showAllTags.value)
+watch(
+  () => tags,
+  () => {
+    showAllTags.value = false
+    hiddenTags.value = 0
+  },
+)
+</script>
+
+<template>
+  <div v-if="tags.length > 0" ref="containerRef" class="Tags">
+    <template v-for="(tag, i) in tags" :key="i">
+      <DocsTag
+        class="Tag"
+        v-bind="tag"
+        :parentOffset="parentOffset"
+        :forceShow="showAllTags"
+        @hidden="tagWasHidden"
+      />
+    </template>
+    <button v-if="showEllipsis" class="Tag button" @click="() => (showAllTags = true)">…</button>
```

Comment:
This approach to hiding excess tags, relying on side effects (DOM property read) in `computed` values within each `DocsTag`, seems fragile. I don't think it will behave responsively if the docs panel is resized.

I propose an alternative:
- Each `DocsTag` renders unconditionally
- Initially, the container has `flex-wrap: nowrap`
- There's a value `haveOverflow: Ref<boolean>`. It is maintained by a resize observer attached to the container; on any size change, the observer updates `haveOverflow.value = elt.clientWidth < elt.scrollWidth`

Reply:
It is not that easy, because `haveOverflow` will constantly flicker. The approach is better though, because we support resizing the doc panel nicely. I implemented most of the new solution, but struggling to place `…` button correctly, so I will include all these changes to the second PR.